pkg/datapath/iptables/iptables.go

```diff
@@ -233,6 +250,179 @@ func RemoveRules() {
+	// TPROXY params
+	ingressProxyMark := fmt.Sprintf("%#08x", linux_defaults.MagicMarkIsToProxy)
+	ingressProxyPort := fmt.Sprintf("%d", proxyPort)
+
+	var err error
+	if option.Config.EnableIPv4 {
+		err = runProg("iptables",
+			ingressProxyRule(cmd, node.GetIPv4AllocRange().String(), l4proto, ingressMarkMatch,
+				ingressProxyMark, ingressProxyPort, name),
+			false)
+	}
+	if err == nil && option.Config.EnableIPv6 {
+		err = runProg("ip6tables",
+			ingressProxyRule(cmd, node.GetIPv6AllocRange().String(), l4proto, ingressMarkMatch,
+				ingressProxyMark, ingressProxyPort, name),
+			false)
+	}
+	return err
+}
+
+func egressProxyRule(cmd, l4Match, markMatch, mark, port, name string) []string {
+	return []string{
+		"-t", "mangle",
+		cmd, ciliumPreMangleChain,
+		"-i", "lxc+",
+		"-p", l4Match,
+		"-m", "mark", "--mark", markMatch,
+		"-m", "comment", "--comment", "cilium: TPROXY to host " + name + " proxy on lxc+",
+		"-j", "TPROXY",
+		"--tproxy-mark", mark,
+		"--on-port", port}
+}
+
+func iptEgressProxyRule(cmd string, l4proto string, proxyPort uint16, name string) error {
+	// Match
+	port := uint32(byteorder.HostToNetwork(proxyPort).(uint16)) << 16
+	egressMarkMatch := fmt.Sprintf("%#08x", linux_defaults.MagicMarkIsToProxy|port)
+	// TPROXY params
+	egressProxyMark := fmt.Sprintf("%#08x", linux_defaults.MagicMarkIsToProxy)
+	egressProxyPort := fmt.Sprintf("%d", proxyPort)
+
+	var err error
+	if option.Config.EnableIPv4 {
+		err = runProg("iptables",
+			egressProxyRule(cmd, l4proto, egressMarkMatch,
+				egressProxyMark, egressProxyPort, name),
+			false)
+	}
+	if err == nil && option.Config.EnableIPv6 {
+		err = runProg("ip6tables",
+			egressProxyRule(cmd, l4proto, egressMarkMatch,
+				egressProxyMark, egressProxyPort, name),
+			false)
+	}
+	return err
+}
+
+func iptRange(cidr *cidr.CIDR) string {
+	start := cidr.IP.Mask(cidr.Mask)
+	end := net.IP(make([]byte, len(cidr.IP)))
+	for i := range cidr.IP {
+		end[i] = cidr.IP[i] | ^cidr.Mask[i]
+	}
+	return start.String() + "-" + end.String()
+}
+
+func installProxyNotrackRules() error {
+	// match traffic to a proxy (upper 16 bits has the proxy port, which is masked out)
+	matchToProxy := fmt.Sprintf("%#08x/%#08x", linux_defaults.MagicMarkIsToProxy, linux_defaults.MagicMarkHostMask)
+	// proxy return traffic has 0 ID in the mask
+	matchProxyReply := fmt.Sprintf("%#08x/%#08x", linux_defaults.MagicMarkIsProxy, linux_defaults.MagicMarkProxyNoIDMask)
+
+	var err error
+	if option.Config.EnableIPv4 {
+		// No conntrack for traffic to proxy
+		err = runProg("iptables", []string{
+			"-t", "raw",
+			"-A", ciliumPreRawChain,
+			// Destination is a local node POD address
+			"!", "-d", node.GetInternalIPv4().String(),
+			"-m", "iprange", "--dst-range", iptRange(node.GetIPv4AllocRange()),
+			"-m", "mark", "--mark", matchToProxy,
+			"-m", "comment", "--comment", "cilium: NOTRACK for proxy traffic",
+			"-j", "NOTRACK"}, false)
```

Comment:
NOTRACK is definitely not widely available. What will break if this is not available?